src/rewrite.ml

```diff
@@ -77,27 +185,10 @@ let handle_rewrite : term -> unit = fun t ->
     | None    ->
         fatal_no_pos "Cannot infer the type of [%a] (given to rewrite)." pp t
   in
-
-  (* Check that the type of [t] is of the form “P (Eq a l r)”. *)
```

Comment:
Similarly, I think it is best to inline the `break_eq` function, as it is very specific. However, the `is_symb` function can be take out. It could even be moved to the `src/terms.ml` file (e.g., right before the definition of `eq`).